symphony/bdk/core/activity/registry.py

```diff
@@ -47,10 +47,18 @@ def register(self, activity: AbstractActivity):
         :param activity: any object inheriting from base :class:`AbstractActivity`
         """
         logger.debug("Registering new activity %s", activity)
+        self._pre_process_activity(activity)
         self._activity_list.append(activity)
 
+    def _pre_process_activity(self, activity: AbstractActivity):
+        if isinstance(activity, SlashCommandActivity):
```

Comment:
I suppose this is useless since it will call the __eq__ function of the correct class (cf. polymorphism).

Reply:
Done in next commit